chain/epoch_manager/src/reward_calculator.rs

```diff
@@ -25,16 +25,18 @@ impl RewardCalculator {
     ) -> (HashMap<AccountId, Balance>, Balance) {
         let mut res = HashMap::new();
         let num_validators = validator_block_chunk_stats.len();
-        let max_inflation = (U256::from(self.max_inflation_rate)
+        let max_inflation = (U256::from(*self.max_inflation_rate.numer() as u64)
             * U256::from(total_supply)
             * U256::from(self.epoch_length)
-            / (U256::from(100) * U256::from(self.num_blocks_per_year)))
+            / (U256::from(self.num_blocks_per_year)
+                * U256::from(*self.max_inflation_rate.denom() as u64)))
         .as_u128();
         let epoch_fee = total_validator_reward;
         let inflation = if max_inflation > epoch_fee { max_inflation - epoch_fee } else { 0 };
         let epoch_total_reward = max(max_inflation, epoch_fee);
-        let epoch_protocol_treasury =
-            epoch_total_reward * u128::from(self.protocol_reward_percentage) / 100;
+        let epoch_protocol_treasury = epoch_total_reward
```

Comment:
Why not use U256 here as well. Do we expect that the `protocol_reward_percentage` numerator always be small enough to fit into u128?